### ISSUE TITLE
`draw_key_rect` fails to get color from outline when `fill` is `NA`

### ISSUE THREAD

**Issue by Ductmonkey** [tidyverse/ggplot2#5385]
#### Problem:
When using `draw_key_rect` with a `geom` that has `fill = NA`, it fails to catch the outline color and returns an invisible (`NA`) rectangle.

#### Minimal reprex:
``` r
library(ggplot2)
ggplot(mtcars, aes(disp, color = factor(cyl))) + geom_density(key_glyph = draw_key_rect)
```

![](https://i.imgur.com/b13Yx9G.png)

<sup>Created on 2023-08-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by clauswilke** [comment]
I think this is actually a bug. The rectangle draw key was meant to pick up the outline color when the fill color isn't given, but that apparently doesn't work. Not sure if it did work in the past and doesn't anymore or if it never worked.

The problem is in this line:
https://github.com/tidyverse/ggplot2/blob/2cd0e96eba7ea2010b07b0d83e43ab12b2cf9eaa/R/legend-draw.R#L66

It is meant to pick `colour` when `fill` is not given, but here `fill` is set to `NA` and `%||%` tests for `NULL`.

There's nothing to change in the documentation, because the `data`, `params`, and `size` arguments to the draw key functions are not meant to be used by the end-user.

**Comment by Ductmonkey** [comment]
Thanks for clarifying, very helpful! Can you migrate this to a bug report then?

**Comment by clauswilke** [comment]
You can just edit the title accordingly. I've added the "bug" label.

**Comment by Ductmonkey** [comment]
Not realizing it was a bug, I had also asked this as a question on StackOverflow. Stefan's answer there may help on this: 
[https://stackoverflow.com/q/76872072/13210554](https://stackoverflow.com/q/76872072/13210554)

**Comment by teunbrand** [comment]
I agree that this is a bug, but there currently isn't a way of distinguishing whether the `fill = NA` comes from the defaults, in which case the colour should override, or it comes the user, in which case it shouldn't.
